editoast/src/views/v2/train_schedule/projection.rs

```diff
@@ -245,15 +245,32 @@ async fn project_path(
             &path_routes,
             &path_blocks,
         );
-        let projection: Option<CachedProjectPathTrainResult> = redis_conn
-            .json_get_ex(&hash, CACHE_PROJECTION_EXPIRATION)
-            .await?;
+        trains_hash_values.insert(train.id, hash);
+        trains_details.insert(train.id, train_details);
+    }
+    let cached_projections: Vec<Option<CachedProjectPathTrainResult>> = redis_conn
+        .json_get_bulk(&trains_hash_values.values().collect::<Vec<_>>())
+        .await?;
+
+    let mut hit_cache: HashMap<i64, CachedProjectPathTrainResult> = HashMap::new();
+    let mut miss_cache = HashMap::new();
+    for ((train_id, train_details), projection) in
+        trains_details.into_iter().zip(cached_projections)
+    {
         if let Some(cached) = projection {
-            hit_cache.insert(train.id, cached);
+            hit_cache.insert(train_id, cached);
         } else {
-            miss_cache.insert(train.id, train_details);
+            miss_cache.insert(train_id, train_details.clone());
         }
     }
+
+    // Reset expire time of hit cache
+    let mut pipe = redis::pipe();
+    for id in hit_cache.keys() {
+        pipe.expire(&trains_hash_values[id], CACHE_PROJECTION_EXPIRATION as i64);
+    }
+    pipe.query_async(&mut redis_conn).await?;
```

Comment:
I feel like this is mostly an implementation detail, nothing really related with the endpoint's logic. Maybe it'd be better to put that away in `redis_utils`? Ideally, I'd say that business logic should just care about an available storage to get objects. The fact that these objects are backed up by a cache solution is not the concern of this function.

Reply:
I agree.
This could be a part of the `json_get_bulk` function.

But if we move to redis's allkeys-lru strategy, is there still a need to manage expiration times ?

Reply:
> But if we move to redis's allkeys-lru strategy, is there still a need to manage expiration times ?

@ElysaSrc helped me setup the policy. So we don't need to manage this anymore :tada: